packages/snaps-execution-environments/src/common/BaseSnapExecutor.test.browser.ts

```diff
@@ -1016,6 +1016,65 @@ describe('BaseSnapExecutor', () => {
     });
   });
 
+  it('reports when outbound requests are made using fetch', async () => {
+    const CODE = `
+      module.exports.onRpcRequest = () => fetch('https://metamask.io').then(res => res.text());
+    `;
+
+    const fetchSpy = spy(globalThis, 'fetch');
+
+    fetchSpy.mockImplementation(async () => {
+      return new Response('foo');
+    });
+
+    const executor = new TestSnapExecutor();
+    await executor.executeSnap(1, MOCK_SNAP_ID, CODE, ['fetch']);
+
+    expect(await executor.readCommand()).toStrictEqual({
+      jsonrpc: '2.0',
+      id: 1,
+      result: 'OK',
+    });
+
+    await executor.writeCommand({
+      jsonrpc: '2.0',
+      id: 2,
+      method: 'snapRpc',
+      params: [
+        MOCK_SNAP_ID,
+        HandlerType.OnRpcRequest,
+        MOCK_ORIGIN,
+        { jsonrpc: '2.0', method: '', params: [] },
+      ],
+    });
+
+    expect(await executor.readCommand()).toStrictEqual({
+      jsonrpc: '2.0',
+      method: 'OutboundRequest',
+    });
+
+    expect(await executor.readCommand()).toStrictEqual({
+      jsonrpc: '2.0',
+      method: 'OutboundResponse',
+    });
+
+    expect(await executor.readCommand()).toStrictEqual({
```

Comment:
I think that test should know which one of these Outbound requests or responses are realted to the `fetch`.

`fetch` can be called multiple times for different URLs or in loop, etc. which would make testing and debugging much more difficult if the request/response patterns are the same as the one used for platform's primary functionality.

Reply:
Would a parameter added to `OutboundRequest` / `OutboundResponse` increase confidence? We could do something like:
```
this.notify({ method: 'OutboundRequest', params: { source: 'fetch' }})
```

Reply:
Yes, exactly something like that. `source` or `origin` or similar.

Reply:
Done! https://github.com/MetaMask/snaps/pull/1756/commits/5bc6d653ea38896dad614228e294cb2e38710f78